val.py

```diff
@@ -156,6 +157,60 @@ def process_batch(detections, labels, iouv):
     return torch.tensor(correct, dtype=torch.bool, device=iouv.device)
 
 
+def get_current_time():
+    current_time = datetime.now()
+    current_time_str = current_time.strftime("%Y-%m-%d %H:%M:%S")  # Format the datetime as a string
+    return current_time_str
+
+
+def exception_handler(func):
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            # Handle the exception here
+            print(f"Exception caught: {e}")
+
+            db_username = kwargs.get('db_username', '')
+            db_name = kwargs.get('db_name', '')
+            db_hostname = kwargs.get('db_hostname', '')
+            run_id = kwargs.get('run_id', 'default_run_id')
+            trained_yolo_model = kwargs.get('trained_yolo_model', 'default')
+            start_time = kwargs.get('start_time', '')
+            # Check if 'skip_evaluation' is provided as an argument in kwargs, and if not, default to False
+            skip_evaluation = kwargs.get('skip_evaluation', False)
+
+            if skip_evaluation:
+                # Validate if database credentials are provided
+                if not db_username or not db_name or not db_hostname:
+                    raise ValueError('Please provide database credentials.')
+
+                # Create a DBConfigSQLAlchemy object
+                db_config = DBConfigSQLAlchemy(db_username, db_hostname, db_name)
+                # Create the database connection
+                db_config.create_connection()
+                # Start the token renewal thread
+                db_config.start_token_renewal_thread()
```

Comment:
Even if we chose this thread solution, the end user should not be the one starting this thread, but it should be handled internally.